MapboxNavigation/LanesView.swift

```diff
@@ -66,31 +66,28 @@ open class LanesView: UIView {
         separatorView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
     }
     
-    func update(for currentLegProgress: RouteLegProgress) {
-        guard let step = currentLegProgress.upComingStep else { return }
-        guard !currentLegProgress.userHasArrivedAtWaypoint else { return }
-        let durationRemaining = currentLegProgress.currentStepProgress.durationRemaining
-        
+    func update(for visualInstruction: VisualInstructionBanner?) {
         clearLaneViews()
         
-        if let allLanes = step.intersections?.first?.approachLanes,
-            let usableLanes = step.intersections?.first?.usableApproachLanes,
-            durationRemaining < RouteControllerMediumAlertInterval {
-            
-            for (i, lane) in allLanes.enumerated() {
-                let laneView = laneArrowView()
-                laneView.lane = lane
-                laneView.maneuverDirection = step.maneuverDirection
-                laneView.isValid = usableLanes.contains(i as Int)
-                stackView.addArrangedSubview(laneView)
-            }
+        guard let tertiaryInstruction = visualInstruction?.tertiaryInstruction,
+                  tertiaryInstruction.containsLaneIndications else {
+                    hide()
+                    return
         }
         
-        if stackView.arrangedSubviews.count > 0 {
-            show()
-        } else {
+        let laneIndications: [LaneIndicationComponent]? = tertiaryInstruction.components.compactMap({ component in
+            guard let lane = component as? LaneIndicationComponent else { return nil }
```

Comment:
This can be written more succinctly as `{ component as? LaneIndicationComponent }`.